stdlib/test/builtin/test_simd.mojo

```diff
@@ -102,6 +102,24 @@ def test_floordiv():
     assert_equal(Float32(2) // Float32(-2), -1)
     assert_equal(Float32(99) // Float32(-2), -50)
 
+    assert_equal(
+        SIMD[DType.int32, 4](2, 4, 1, 5) // Int32(2),
```

Comment:
isn't this actually calling `__floordiv__` instead of `__rfloordiv__`? Since the compiler will try `__floordiv__` first if the two elements are both SIMD. This raises the question, in what situation is this implementation of `__rfloordiv__` useful? In python, it's only used to handle other types. 

Maybe there is some compiler logic I'm missing here?

Reply:
Correct that test and the one right below it call __floordiv__(). I added those tests to have more test coverage for __floordiv__(). I can exclude them in this PR and add them in another PR if its misleading.

Reply:
Then is which situation is it possible to call the function`__rfloordiv__` that was just added?

Reply:
It can happen if there is some implicit conversion. It can also be required by some traits (e.g. `CeilDivable`). See https://github.com/modularml/mojo/pull/2414#issuecomment-2098934194